dcrtimed/backend/filesystem/filesystem.go

```diff
@@ -748,6 +750,90 @@ func (fs *FileSystem) GetTimestamps(timestamps []int64) ([]backend.TimestampResu
 	return gtmes, nil
 }
 
+// Get the last n digests in the added to the Backend
+func (fs *FileSystem) LastDigests(n int32) ([]backend.GetResult, error) {
+
+	if n > fs.maxDigestsNumber {
+		return nil, fmt.Errorf("Invalid number %d of digests requested. Max is: %d", n, fs.maxDigestsNumber)
+	}
+
+	files, err := os.ReadDir(fs.root)
+	if err != nil {
+		return nil, err
+	}
+
+	// We need to be read locked from here on out.
+	fs.RLock()
+	defer fs.RUnlock()
+
+	results := make([]backend.GetResult, 0)
+
+	// Reverse slice so we look at the most recent files first
```

Comment:
If this was a bounded slice in size then reversing it wouldn't be a big deal, but isn't it possible for this to contain tens of thousands or even hundreds of thousands of elements?  Wouldn't it make more sense to just iterate starting from the end of the slice instead of reversing the slice?